handler/src/main/java/io/netty/handler/pcap/PcapWriteHandler.java

```diff
@@ -0,0 +1,522 @@
+    private final OutputStream outputStream;
+
+    /**
+     * {@code true} if we want to capture packets with zero bytes else {@code false}.
+     */
+    private final boolean captureZeroByte;
+
+    /**
+     * {@code true} if we want to write Pcap Global Header on initialization of
+     * {@link PcapWriter} else {@code false}.
+     */
+    private final boolean writePcapGlobalHeader;
+
+    /**
+     * TCP Sender Segment Number.
+     * It'll start with 1 and keep incrementing with number of bytes read/sent.
+     */
+    private int sendSegmentNumber = 1;
+
+    /**
+     * TCP Receiver Segment Number.
+     * It'll start with 1 and keep incrementing with number of bytes read/sent.
+     */
+    private int receiveSegmentNumber = 1;
+
+    /**
+     * Source Address
+     */
+    private InetSocketAddress srcAddr;
+
+    /**
+     * Destination Address
+     */
+    private InetSocketAddress dstAddr;
+
+    /**
+     * Create new {@link PcapWriteHandler} Instance.
+     * {@code captureZeroByte} is set to {@code false} and
+     * {@code writePcapGlobalHeader} is set to {@code true}.
+     *
+     * @param outputStream OutputStream where Pcap data will be written
+     */
+    public PcapWriteHandler(OutputStream outputStream) {
+        this(outputStream, false, true);
+    }
+
+    /**
+     * Create new {@link PcapWriteHandler} Instance
+     *
+     * @param outputStream          OutputStream where Pcap data will be written
+     * @param captureZeroByte       Set to {@code true} to enable capturing packets with empty (0 bytes) payload.
+     *                              Otherwise, if set to {@code false}, empty packets will be filtered out.
+     * @param writePcapGlobalHeader Set to {@code true} to write Pcap Global Header on initialization.
+     *                              Otherwise, if set to {@code false}, Pcap Global Header will not be written
+     *                              on initialization. This could when writing Pcap data on a existing file where
+     *                              Pcap Global Header is already present.
+     */
+    public PcapWriteHandler(OutputStream outputStream, boolean captureZeroByte, boolean writePcapGlobalHeader) {
+        this.outputStream = outputStream;
+        this.captureZeroByte = captureZeroByte;
+        this.writePcapGlobalHeader = writePcapGlobalHeader;
+    }
+
+    @Override
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
+
+        ByteBufAllocator byteBufAllocator = ctx.alloc();
+
+        /*
+         * If `writePcapGlobalHeader` is `true`, we'll write Pcap Global Header.
+         */
+        if (writePcapGlobalHeader) {
+
+            ByteBuf byteBuf = byteBufAllocator.buffer();
+            try {
+                this.pCapWriter = new PcapWriter(this.outputStream, byteBuf);
+            } catch (IOException ex) {
+                ctx.fireExceptionCaught(ex);
+            } finally {
+                byteBuf.release();
+            }
```

Comment:
this doesn't look right... if we see an `IOException` here `pCapWriter` will be null and so we will fail with an NPE later on. I think what you should do is just release the `ByteBuf`, close the channel and let the exception bubble up.

Reply:
Should I close the channel every time I get exception?

Reply:
Should I close the channel every time I get an exception?

Reply:
not really... its more like if you not close it here we will be in an "invalid" state. Alternative you could also remove this handler and log.

Reply:
Done, PTAL. :)